c10/core/SymbolicIntNode.h

```diff
@@ -13,7 +13,53 @@ class C10_API SymbolicIntNode
  public:
   c10::SymInt toSymInt();
   virtual ~SymbolicIntNode(){};
-  virtual std::ostream& operator<<(std::ostream& os) {
+  // these could be pure virtual when we implement LTC versions
+  virtual std::shared_ptr<SymbolicIntNode> add(
+      std::shared_ptr<SymbolicIntNode> other) {
```

Comment:
You aren't taking ownership of the argument, so I would suggest taking a constant reference

Reply:
right!